src/actions/InvoiceActions.js

```diff
@@ -47,7 +54,23 @@ export const invoiceUpdateById = (invoice) => {
     .catch(error => {
       console.warn('The invoice was not updated', error);
     });
-  dispatch({ type: INVOICE_CREATE, payload: { uid } });//Todo revisar para que sirve el dispatch
+    dispatch({ type: INVOICE_UPDATE_SUCCESS, payload: invoice, uid });
+  }
+}
+
+export const invoiceUpdateRiderById = (invoice) => {
+  const {uid} = invoice;
+  delete invoice.uid;
+  return (dispatch) => {
+    firebase.firestore().collection('invoices').doc(uid)
+    .update(invoice)
+    .then(() => {
+      console.info(`Invoice Rider updated with id ${uid}`);
+    })
+    .catch(error => {
+      console.warn('The invoice rider was not updated', error);
+    });
+    dispatch({ type: INVOICE_RIDER_UPDATE_SUCCESS, payload: invoice, uid });
```

Comment:
I am no sure if this works since that you are sending the uid out of the payload, please take a look. Also I think that the dispatch should be after the `then` since that you are waiting the update of the invoice